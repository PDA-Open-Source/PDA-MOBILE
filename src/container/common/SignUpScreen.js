/**
 * pda
 * SignUpScreen.js
 * @author PDA
 * @description Created on 07/06/2019
 * Copyright Â© 2019 pda. All rights reserved.
 */
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  userSignUp,
  getCountryCodeList
} from '../../redux/actions/UserActions';
import SignUp from '../../components/screens/common/SignUp';
const mapStateToProps = (state) => {
  const { isInternetConnectivityAvailable } = state.appState;
  return {
    isNetworkConnected : isInternetConnectivityAvailable,
    countryList: state.userState.countryList,
    countryCodeWithCountryList: state.userState.countryCodeWithCountryList
  };
};


const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    userSignUp,
    getCountryCodeList
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);
