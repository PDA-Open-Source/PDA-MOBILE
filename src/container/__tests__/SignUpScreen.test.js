import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import SignUpScreen from '../SignUpScreen';

const mockStore = configureMockStore();
const store = mockStore({appState:{
  isInternetConnectivityAvailable: true
}});
jest.mock('react-native-device-info', () => {
  return {
    hasNotch: () => false
  };
});

describe('SignUp Screen rendering',()=>{
  it('renders the SignUp screen ', async () => {
    const tree = renderer.create(
      <Provider store={store}>
        <SignUpScreen />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});