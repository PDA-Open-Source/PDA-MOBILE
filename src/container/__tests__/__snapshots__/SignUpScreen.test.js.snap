// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Screen rendering renders the SignUp screen  1`] = `
<View
  pointerEvents={null}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/Images/background.jpeg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <RCTScrollView
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.6)",
          "flex": 1,
          "padding": 39.23529411764706,
        }
      }
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "padding": 39.23529411764706,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/Images/logo.png",
                }
              }
              style={
                Object {
                  "height": 156.94117647058823,
                  "width": 156.94117647058823,
                }
              }
              testID="logo"
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                },
                Object {
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 40.85714285714286,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {
                  "paddingVertical": 19.61764705882353,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sign Up
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "color": "#fff",
                    "height": 50,
                    "width": 100,
                  }
                }
              >
                <RCTPicker
                  items={
                    Array [
                      Object {
                        "label": "Mr.",
                        "textColor": undefined,
                        "value": "Mr.",
                      },
                      Object {
                        "label": "Miss.",
                        "textColor": undefined,
                        "value": "Miss.",
                      },
                      Object {
                        "label": "Ms.",
                        "textColor": undefined,
                        "value": "Ms.",
                      },
                      Object {
                        "label": "Mrs.",
                        "textColor": undefined,
                        "value": "Mrs.",
                      },
                    ]
                  }
                  onChange={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  selectedIndex={0}
                  style={
                    Array [
                      Object {
                        "height": 216,
                      },
                      undefined,
                    ]
                  }
                  testID="titlePicker"
                />
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "height": 1,
                    "marginLeft": 9.808823529411764,
                    "marginRight": 39.23529411764706,
                    "marginVertical": -5,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View>
                <TextInput
                  accessibilityLabel="firstName_textInput"
                  allowFontScaling={true}
                  autoCapitalize="none"
                  keyboardType="default"
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="First Name*"
                  placeholderTextColor="#fff"
                  returnKeyType="next"
                  style={
                    Array [
                      Object {
                        "color": "#fff",
                        "marginTop": 9.808823529411764,
                      },
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 23.57142857142857,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "fontFamily": "Roboto-Light",
                      },
                      undefined,
                    ]
                  }
                  testID="firstName_textInput"
                  underlineColorAndroid="#fff"
                  value=""
                />
              </View>
            </View>
          </View>
          <View>
            <TextInput
              accessibilityLabel="lastName_textInput"
              allowFontScaling={true}
              autoCapitalize="none"
              keyboardType="default"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Last Name*"
              placeholderTextColor="#fff"
              returnKeyType="next"
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "marginTop": 9.808823529411764,
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 23.57142857142857,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontFamily": "Roboto-Light",
                  },
                  undefined,
                ]
              }
              testID="lastName_textInput"
              underlineColorAndroid="#fff"
              value=""
            />
          </View>
          <View>
            <TextInput
              accessibilityLabel="email_textInput"
              allowFontScaling={true}
              autoCapitalize="none"
              keyboardType="email-address"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Email Id*"
              placeholderTextColor="#fff"
              returnKeyType="next"
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "marginTop": 9.808823529411764,
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 23.57142857142857,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontFamily": "Roboto-Light",
                  },
                  undefined,
                ]
              }
              testID="email_textInput"
              underlineColorAndroid="#fff"
              value=""
            />
          </View>
          <View>
            <TextInput
              accessibilityLabel="password_textInput"
              allowFontScaling={true}
              autoCapitalize="none"
              keyboardType="default"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Password*"
              placeholderTextColor="#fff"
              returnKeyType="next"
              secureTextEntry={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "marginTop": 9.808823529411764,
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 23.57142857142857,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontFamily": "Roboto-Light",
                  },
                  undefined,
                ]
              }
              testID="password_textInput"
              underlineColorAndroid="#fff"
              value=""
            />
          </View>
          <View>
            <TextInput
              accessibilityLabel="retypePassword_textInput"
              allowFontScaling={true}
              autoCapitalize="none"
              keyboardType="default"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Retype Password*"
              placeholderTextColor="#fff"
              returnKeyType="done"
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "marginTop": 9.808823529411764,
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 23.57142857142857,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "fontFamily": "Roboto-Light",
                  },
                  undefined,
                ]
              }
              testID="retypePassword_textInput"
              underlineColorAndroid="#fff"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "marginVertical": 78.47058823529412,
              }
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4283283116,
                  4280723098,
                ]
              }
              endPoint={
                Object {
                  "x": 0.5,
                  "y": 1,
                }
              }
              locations={null}
              startPoint={
                Object {
                  "x": 0.5,
                  "y": 0,
                }
              }
              style={
                Array [
                  Object {
                    "backgroundColor": "#26A69A",
                    "borderRadius": 5,
                    "marginVertical": 19.61764705882353,
                    "paddingVertical": 9.808823529411764,
                    "width": 392.35294117647055,
                  },
                  Object {
                    "alignItems": "center",
                    "borderRadius": 5,
                  },
                ]
              }
            >
              <View
                accessible={true}
                onLayout={[Function]}
                onPress={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onRippleAnimation={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                rippleCentered={false}
                rippleColor="#546E7A"
                rippleDuration={400}
                rippleFades={true}
                rippleOpacity={0.3}
                rippleSequential={false}
                rippleSize={0}
                style={
                  Object {
                    "borderRadius": 5,
                    "paddingLeft": 25,
                    "paddingRight": 25,
                    "paddingVertical": 10,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      undefined,
                      Array [
                        Object {
                          "color": "#fff",
                          "fontWeight": "bold",
                          "paddingHorizontal": 78.47058823529412,
                        },
                        Object {
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 25.142857142857142,
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ],
                      Object {
                        "fontFamily": "Roboto-Regular",
                        "fontSize": 23.57142857142857,
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  SIGN UP
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "bottom": 0,
                        "left": 0,
                        "overflow": "hidden",
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                      Object {
                        "borderRadius": 0,
                      },
                    ]
                  }
                />
              </View>
            </BVLinearGradient>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "justifyContent": "center",
                "marginBottom": 58.85294117647059,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                  },
                  Object {
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 23.57142857142857,
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {
                    "paddingVertical": 19.61764705882353,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Already have account?
              <Text
                onPress={[Function]}
                style={
                  Object {
                    "color": "#4DB6AC",
                  }
                }
                testID="Tologin_Text"
              >
                 
                Login now
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
